chain-core/src/property.rs

```diff
@@ -60,11 +60,17 @@ pub trait Header: Serialize + Deserialize {
     /// The block date.
     type Date: BlockDate;
 
+    /// the type associated to the version of a block
+    type Version;
```

Comment:
Might want to add an empty tag trait `Version` and bind this associated type with it, just like we do for `Id` and `Date`.